Code/GraphMol/SmilesParse/SmilesWrite.cpp

```diff
@@ -717,4 +717,22 @@ std::string MolFragmentToSmiles(const ROMol &mol,
   mol.setProp(common_properties::_smilesAtomOutputOrder, atomOrdering, true);
   return res;
 }  // end of MolFragmentToSmiles()
+
+std::string MolFragmentToCXSmiles(const ROMol &mol,
+                                  const std::vector<int> &atomsToUse,
+                                  const std::vector<int> *bondsToUse,
+                                  const std::vector<std::string> *atomSymbols,
+                                  const std::vector<std::string> *bondSymbols,
+                                  bool doIsomericSmiles, bool doKekule,
+                                  int rootedAtAtom, bool canonical,
+                                  bool allBondsExplicit, bool allHsExplicit) {
+  auto res = MolFragmentToSmiles(
+      mol, atomsToUse, bondsToUse, atomSymbols, bondSymbols, doIsomericSmiles,
+      doKekule, rootedAtAtom, canonical, allBondsExplicit, allHsExplicit);
+  auto cxext = SmilesWrite::getCXExtensions(mol);
```

Comment:
I’m missing something here, how can the extensions map from the mol to res (the fragment) indices are different and so on.

Reply:
Should this be

auto cxext = SmilesWrite::getCXExtensions(res);

Reply:
Yeah, it seems odd, but it is right. `getCXExtensions()` uses the `_smilesAtomOutputOrder` property to figure out what to generate (and what indices to use):
https://github.com/rdkit/rdkit/blob/2dfc222513acaa8012953820818d3d8380b1f345/Code/GraphMol/SmilesParse/CXSmilesOps.cpp#L514